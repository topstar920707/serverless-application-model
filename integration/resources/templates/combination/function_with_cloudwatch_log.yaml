Resources:
  MyLambdaFunction:
    Type: 'AWS::Serverless::Function'
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      CodeUri: ${codeuri}
      MemorySize: 128
      Events:
        LogProcessor:
          Type: CloudWatchLogs
          Properties:
            LogGroupName:
              Ref: CloudWatchLambdaLogsGroup
            FilterPattern: My filter pattern
            
  CloudWatchLambdaLogsGroup:
    Type: AWS::Logs::LogGroup
    Properties:
      RetentionInDays: 7