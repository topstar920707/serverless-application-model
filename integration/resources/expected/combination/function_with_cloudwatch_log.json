[
  { "LogicalResourceId":"MyLambdaFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyLambdaFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyLambdaFunctionLogProcessorPermission", "ResourceType":"AWS::Lambda::Permission" },
  { "LogicalResourceId":"CloudWatchLambdaLogsGroup", "ResourceType":"AWS::Logs::LogGroup" },
  { "LogicalResourceId":"MyLambdaFunctionLogProcessor", "ResourceType":"AWS::Logs::SubscriptionFilter" }
]